tests/test_spark_users.py

```diff
@@ -0,0 +1,137 @@
+import logging
+
+import pytest
+
+import sdk_cmd
+import sdk_tasks
+import shakedown
+import spark_utils as utils
+import docker_utils
+
+log = logging.getLogger(__name__)
+
+SERVICE_NAME = "test-users"
+DISPATCHER_ZK = "spark_mesos_dispatcher" + SERVICE_NAME
+
+LIST_USERS_CMD = "ps aux | grep java | cut -d' ' -f1 | uniq | tail -1"
+
+
+@pytest.fixture(scope='module')
+def configure_security():
+    yield from utils.spark_security_session()
+
+
+@pytest.fixture()
+def setup_spark(configure_security, configure_universe, use_ucr_containerizer, user):
+    options = {
+        "service": {
+            "name": SERVICE_NAME,
+            "user": user,
+            "UCR_containerizer": use_ucr_containerizer
+        }
+    }
+
+    try:
+        utils.upload_dcos_test_jar()
+        utils.require_spark(service_name=SERVICE_NAME, additional_options=options, zk=DISPATCHER_ZK)
+        yield
+    finally:
+        utils.teardown_spark(service_name=SERVICE_NAME, zk=DISPATCHER_ZK)
+
+
+@pytest.mark.sanity
+@pytest.mark.parametrize('user,use_ucr_containerizer,use_ucr_for_spark_submit', [
+    ("nobody", True, True),
+    ("nobody", True, False),
+    ("nobody", False, True),
```

Comment:
Do we want to test scenarios where users launch dispatchers with docker but submit jobs with UCR? I guess it doesn't hurt but is this realistic?

Reply:
By default, Dispatcher is deployed using Docker containerizer. From here we can think of the next scenarios:
- a user submits a job (`False`, `False`)
- a user discovers UCR option, updates Dispatcher config, but keeps submitting jobs as before (`True`, `False`)
- user discovers spark-submit option `spark.mesos.containerizer=mesos` but runs Dispatcher with default config (`False`, `True`)
- a user is experienced and enables UCR in both Dispatcher and jobs (`True`, `True`) 

The whole goal of this exhaustive testing is to make sure we don't miss some not-so-obvious combination of containerizers in which user propagation breaks.